flags.go

```diff
@@ -59,18 +66,48 @@ func (b csvFileBackend) Refresh() (map[string]Flag, error) {
 	return readFile(b.filename, "csv", parseFlagsCSV)
 }
 
+// for csv file, update time looks at last modified time on filesystem
+func (b csvFileBackend) UpdateTime() (time.Time, error) {
+	f, err := os.Open(b.filename)
+	if err != nil {
+		return time.Time{}, err
+	}
+	defer f.Close()
+	statinfo, err := f.Stat()
+	if err != nil {
+		return time.Time{}, err
+	}
+	return statinfo.ModTime(), nil
+}
+
+// for json file, update time comes from json directly
+func (b jsonFileBackend) UpdateTime() (time.Time, error) {
+	f, err := os.Open(b.filename)
+	if err != nil {
+		return time.Now(), err
+	}
+	defer f.Close()
+	json, err := parseJSON(f)
+	return time.Unix(int64(json.UpdatedTime), 0), nil
+}
+
 type Flag struct {
 	Name string
 	Rate float64
 }
 
 type JSONFormat struct {
-	Flags []Flag `json:"flags"`
+	Flags       []Flag  `json:"flags"`
+	UpdatedTime float64 `json:"updated"`
 }
 
 var flags = map[string]Flag{}
 var flagsMtx = sync.RWMutex{}
 
+var lastUpdate time.Time
+var updateMtx = sync.RWMutex{}
```

Comment:
I feel like it's ok to use the same mutex for both updates and flags.

Reply:
Done